ui/src/utils/time.ts

```diff
@@ -13,3 +13,24 @@ export const formatTimestamp = (timestamp: string | Date) => {
 
   return date.toLocaleString(undefined, options);
 };
+
+// Round to nearest time unit e.g.:
+// nearest minute (timeUnitMs === 60000)
+// nearest 5 minutes (timeUnitMs === 300000)
+// nearest 15 minutes (timeUnitMs === 900000)
+export const roundToNearest = (
+  date: Date,
+  timeUnitMs: number,
+  roundMethod: "round" | "ceil" | "floor",
+) => {
+  let roundFn;
+  if (roundMethod === "round") {
+    roundFn = Math.round;
+  } else if (roundMethod === "ceil") {
+    roundFn = Math.ceil;
+  } else {
+    roundFn = Math.floor;
+  }
+
+  return new Date(roundFn(date.getTime() / timeUnitMs) * timeUnitMs);
+};
```

Comment:
How about something like this? (haven't tested)

```suggestion
export const roundToNearest = (
  date: Date,
  timeUnitMs: number,
  roundMethod: "round" | "ceil" | "floor",
): Date => {
  const roundingFn = {
    round: Math.round,
    ceil: Math.ceil,
    floor: Math.floor,
  }[roundMethod];

  const timestamp = date.getTime();
  const roundedTimestamp = roundingFn(timestamp / timeUnitMs) * timeUnitMs;

  return new Date(roundedTimestamp);
};
```